storage/store_test.go

```diff
@@ -1106,3 +1107,49 @@ func TestRaftNodeID(t *testing.T) {
 		}()
 	}
 }
+
+// lookupAndAddRandomRanges adds numRanges ranges with randomly
+// generated start/end keys. Before adding a range, looks up if there is
+// any range that has the same key.
+func lookupAndAddRandomRanges(numRanges int, b *testing.B) {
+	store, _, stopper := createTestStore(b)
+	defer stopper.Stop()
+
+	// Remove range 1.
+	rng1, err := store.GetRange(1)
+	if err != nil {
+		b.Fatal(err)
+	}
+	if err := store.RemoveRange(rng1); err != nil {
+		b.Fatal(err)
+	}
+
+	for i := 0; i < numRanges; i++ {
+		// Start from 2 since Raft ID 1 has already been assigned.
+		raftID := int64(i)
+		for {
+			start := rand.Int63()
+			startKey := proto.Key(fmt.Sprintf("a%d", start))
+			endKey := proto.Key(fmt.Sprintf("a%d", start+1))
+			if store.LookupRange(startKey, endKey) == nil {
+				rng := createRange(store, raftID, startKey, endKey)
+				if err := store.AddRange(rng); err != nil {
+					b.Fatal(err)
+				}
+				break
+			}
+			// Try again with a different key.
+		}
+	}
+}
+
+func benchmarkLookupAndAddRange(numRanges int, b *testing.B) {
+	defer leaktest.AfterTest(b)
+	for i := 0; i < b.N; i++ {
```

Comment:
Rather than looping on `b.N` here, I think you should change the loop in `lookupAndAddRandomRanges` to loop on `b.N` instead of passing in `numRanges`.